evaluator/src/main/kotlin/DependencyRule.kt

```diff
@@ -45,13 +45,13 @@ class DependencyRule(
     val dependency: PackageReference,
 
     /**
-     * The ancestors of the [dependency] in the dependency tree. The first entry is the root of the tree, the last entry
-     * (at [level] - 1) is the direct parent.
+     * The ancestors of the [dependency] in the dependency tree, sorted from farthest to closest: The first entry is the
+     * direct dependency of a project, the last entry (at index [level] - 1) is the direct parent of this [dependency].
```

Comment:
How about removing "(at index [level] - 1)" as "the last entry" already describes it sufficiently?

Reply:
I liked it, as it emphasizes that the number of entries in `ancestors` equals the `level`.

Reply:
Maybe write it as "(at index `level - 1`)".

Reply:
You could make it even clearer by adding: "If the list is empty it means that this dependency is a direct dependency."